qcodes/measurement.py

```diff
@@ -0,0 +1,733 @@
+        name=None,
+        label=None,
+        unit=None,
+    ):
+        """Perform a single measurement of a Parameter, function, etc.
+
+
+        Args:
+            measurable: Item to measure. Can be one of the following:
+                Parameter
+                Callable function/method, which should either perform a nested
+                    Measurement, or return a dict.
+                    In the case of returning a dict, all the key/value pairs
+                    are grouped together.
+                float, int, bool, array
+            name: Optional name for measured element or data group.
+                If the measurable is a float, int, bool, or array, the name is
+                mandatory.
+                Otherwise, the default name is used.
+            label: Optional label, is ignored if measurable is a Parameter or callable
+            unit: Optional unit, is ignored if measurable is a Parameter or callable
+
+        Returns:
+            Return value of measurable
+        """
+        # TODO add label, unit, etc. as kwargs
+        if not self.is_context_manager:
+            raise RuntimeError(
+                "Must use the Measurement as a context manager, "
+                "i.e. 'with Measurement(name) as msmt:'"
+            )
+        elif self.is_stopped:
+            raise SystemExit("Measurement.stop() has been called")
+        elif threading.current_thread() is not Measurement.measurement_thread:
+            raise RuntimeError(
+                "Cannot measure while another measurement is already running "
+                "in a different thread."
+            )
+
+        if self != Measurement.running_measurement:
+            # Since this Measurement is not the running measurement, it is a
+            # DataGroup in the running measurement. Delegate measurement to the
+            # running measurement
+            return Measurement.running_measurement.measure(measurable, name=name)
+
+        # Code from hereon is only reached by the primary measurement,
+        # i.e. the running_measurement
+
+        # Wait as long as the measurement is paused
+        while self.is_paused:
+            sleep(0.1)
+
+        # TODO Incorporate kwargs name, label, and unit, into each of these
+        if isinstance(measurable, Parameter):
+            result = self._measure_parameter(measurable, name=name)
+        elif isinstance(measurable, MultiParameter):
+            result = self._measure_multi_parameter(measurable, name=name)
+        elif callable(measurable):
+            result = self._measure_callable(measurable, name=name)
+        elif isinstance(measurable, (float, int, bool, np.ndarray)):
+            result = self._measure_value(measurable, name=name)
+        else:
+            raise RuntimeError(
+                f"Cannot measure {measurable} as it cannot be called, and it "
+                f"is not an int, float, bool, or numpy array."
+            )
```

Comment:
This code shouldn't be reached due to the typing of the argument. Not sure if it's good to keep the error message or not.